### DISCUSSION TITLE
Can one configure the json key order of the generated content json?

### DISCUSSION THREAD

**Discussion by liufuyang** [decaporg/decap-cms#6760] [Q&A]
We use github as the backend and have scripts managing the content jsons generated from decap-cms. Currently, in our script, we generate the jsons with keys in the order of defined in the `config.yml` file.

The issue we have is when we review the changes either done via out script (or afterwards changes done via the decap-cms UI), it is difficult to know what values got changed as the PR shows a lot of moved fields.

If the current decap-cms's output json key order is not configurable, then do we know what order it follows?

Thank you.

**Comment by martinjagodic** [comment]
I converted this to a discussion as it is a question, not a feature request. I don't know the answer though.

**Reply by liufuyang** [reply]
Thank you. Will you or perhaps someone check on this later?

**Comment by paulgirard** [comment]
The answer is : the order yielded by `Object.keys`. 
It's a stable order although totally not predictable at least to me.

Here is Decap serialization code: https://github.com/decaporg/decap-cms/blob/2f5c54d96d3eaf657d0b75a0a2578073731907a7/packages/decap-cms-core/src/formats/json.ts#L7

Here is the documentation about JSON stringify key order : 
> Properties are visited using the same algorithm as [Object.keys()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys), which has a well-defined order and is stable across implementations. For example, JSON.stringify on the same object will always produce the same string, and JSON.parse(JSON.stringify(obj)) would produce an object with the same key ordering as the original (assuming the object is completely JSON-serializable).
source :https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify
